src/dd/dd.rs

```diff
@@ -0,0 +1,640 @@
+            result += "[swab]";
+        }
+        if self.flags & Self::NOERROR != 0 {
+            result += "[noerror]";
+        }
+        if self.flags & Self::NOTRUNC != 0 {
+            result += "[notrunc]";
+        }
+        if self.flags & Self::SYNC != 0 {
+            result += "[sync]";
+        }
+        result
+    }
+
+    const BLOCK: FlagT = 0b00000001;
+    const UNBLOCK: FlagT = 0b00000010;
+    const LCASE: FlagT = 0b00000100;
+    const UCASE: FlagT = 0b00001000;
+    const SWAB: FlagT = 0b00010000;
+    const NOERROR: FlagT = 0b00100000;
+    const NOTRUNC: FlagT = 0b01000000;
+    const SYNC: FlagT = 0b10000000;
+    const INVALID_FLAG: FlagT = 0b1_00000000;
+}
+
+impl fmt::Display for Conf {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            f,
+            "if: {if_}\n\
+             of: {of_}\n\
+             ibs: {ibs}\n\
+             obs: {obs}\n\
+             cbs: {cbs}\n\
+             skip: {skip}\n\
+             seek: {seek}\n\
+             count: {count}\n\
+             flags: {flags}\n",
+            if_ = self.inp_name,
+            of_ = self.out_name,
+            ibs = self.inp_block_size,
+            obs = self.out_block_size,
+            cbs = self.cnv_block_size,
+            skip = self.skip,
+            seek = self.seek,
+            count = self.count,
+            flags = self.str_flags()
+        )
+    }
+}
+
+pub fn parse_args() -> Conf {
+    let mut cf = Conf::default();
+    for arg in env::args().skip(1) {
+        let pair: Vec<&str> = arg.splitn(2, '=').collect();
+        if pair.len() == 2 {
+            let (key, val) = (pair[0], pair[1]);
+            match key {
+                "if" => cf.set_if(val),
+                "of" => cf.set_of(val),
+                "ibs" => match get_u(val) {
+                    Some(m) => {
+                        cf.inp_block_size = m;
+                    }
+                    _ => err(format!("invalid {}: {}", key, val)),
+                },
+                "obs" => match get_u(val) {
+                    Some(m) => {
+                        cf.out_block_size = m;
+                    }
+                    _ => err(format!("invalid {}: {}", key, val)),
+                },
+                "bs" => match get_u(val) {
+                    Some(m) => {
+                        cf.inp_block_size = m;
+                        cf.out_block_size = m;
+                    }
+                    _ => err(format!("invalid {}: {}", key, val)),
+                },
+                "cbs" => match get_u(val) {
+                    Some(m) => {
+                        cf.cnv_block_size = m;
+                    }
+                    _ => err(format!("invalid {}: {}", key, val)),
+                },
+                "skip" => match get_u(val) {
+                    Some(m) => {
+                        cf.skip = m;
+                    }
+                    _ => err(format!("invalid {}: {}", key, val)),
```

Comment:
You might want to make a function or something to do this sort of stuff.  Most of the `match`es seem pretty similar.

Reply:
Did.